### PR TITLE
chore(docker): Upgrade ScanCode to version v32.1.0

### PR DESCRIPTION
**Pull request by Etsija**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
-RUN apt install -y python3-pip
-RUN pip3 install scancode-toolkit==32.0.4
+# Install required dependencies
+RUN apt-get update && \
+    apt-get install -y python3-venv python3-pip && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Create a virtual environment for Python
+RUN python3 -m venv /venv
```
Comment (sschuberth) [inline Dockerfile:17]: Why is this needed instead of just calling `pip3` explicitly like before?

Comment (Etsija) [reply Dockerfile:17]: Because without it, we get
```
7.160 error: externally-managed-environment
7.160
7.160 × This environment is externally managed
7.160 ╰─> To install Python packages system-wide, try apt install
7.160     python3-xyz, where xyz is the package you are trying to
7.160     install.
7.160
7.160     If you wish to install a non-Debian-packaged Python package,
7.160     create a virtual environment using python3 -m venv path/to/venv.
7.160     Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
7.160     sure you have python3-full installed.
7.160
7.160     If you wish to install a non-Debian packaged Python application,
7.160     it may be easiest to use pipx install xyz, which will manage a
7.160     virtual environment for you. Make sure you have pipx installed.
7.160
7.160     See /usr/share/doc/python3.11/README.venv for more information.
7.160
7.160 note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
7.160 hint: See PEP 668 for the detailed specification.
------
Dockerfile:21
--------------------
  20 |     # Install required Python package
  21 | >>> RUN apt-get update && \
  22 | >>>     apt-get install -y python3-pip && \
  23 | >>>     pip3 install scancode-toolkit==32.1.0 && \
  24 | >>>     apt-get clean && \
  25 | >>>     rm -rf /var/lib/apt/lists/*
  26 |
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get update &&     apt-get install -y python3-pip &&     pip3 install scancode-toolkit==32.1.0 &&     apt-get clean &&     rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 1
```

Comment (Etsija) [reply Dockerfile:17]: I am really not familiar of Python's library handling peculiarities, so don't understand what's going on there.

Comment (Etsija) [reply Dockerfile:17]: With Node 18.15.0, all it took was to use the same old Dockerfile, and upgrade ScanCode in it.  But we really should be able to upgrade Node as well, as 20.10.0 is the version used throughout DOS.

Comment (sschuberth) [reply Dockerfile:17]: > but it all stems from upgrading the Node from 18.15.0 to 20.10.0.

I wonder how that's related to the Python environment, though. But I agree Node should be updated as well.

Maybe @haikoschol as the Python guy knows where the [above message](https://github.com/doubleopen-project/dos/pull/765#discussion_r1542532708) comes from?

Comment (sschuberth) [reply Dockerfile:17]: In any case, the motivation to switch to using a `venv` should be documented in the commit message, please.

Comment (haikoschol) [reply Dockerfile:17]: > > but it all stems from upgrading the Node from 18.15.0 to 20.10.0.
> 
> I wonder how that's related to the Python environment, though. But I agree Node should be updated as well.
> 
> Maybe @haikoschol as the Python guy knows where the [above message](https://github.com/doubleopen-project/dos/pull/765#discussion_r1542532708) comes from?

The message comes from a relatively recent [feature in Python packaging](https://packaging.python.org/en/latest/specifications/externally-managed-environments/#externally-managed-environments) to prevent users from inadvertently breaking OS managed Python packages by installing packages globally with tools like `pip`.

This change switches to a newer Node **image**, as opposed to installing a newer Node version on the same image. I assume that the previous one used a Debian or Ubuntu version as base image that did not include this `EXTERNALLY-MANAGED` marker file and the Node 20 image does include it.

So the issue that this message points at already existed before, but the mechanism to catch it and require explicit override did not.

Comment (Etsija) [reply Dockerfile:17]: @haikoschol Do you have any suggestions for changing in the Dockerfile, as I'm _really_ no Python guy?

Comment (haikoschol) [reply Dockerfile:17]: > @haikoschol Do you have any suggestions for changing in the Dockerfile, as I'm _really_ no Python guy? The link you provided seems to suggest that using a virtual environment is OK in this situation.

Yes, I would say creating a virtualenv is the right approach. Just add something to the commit message to the effect of: "Scancode and its' dependencies are now installed in a virtual environment to avoid the potential for overwriting globally installed Python packages."

**Review by lamppu — APPROVED** [review]
